### PR TITLE
[vcpkg-cmake] New options DISABLE_DEFAULT_PKGCONFIG_PATH and MORE_DEBUG_INFO

### PR DESCRIPTION
**Pull request by JackBoosY**
Since I met the pkgconfig issue twice(one is in https://github.com/microsoft/vcpkg/pull/24969#discussion_r920340040),
we should force `PKG_CONFIG_USE_CMAKE_PREFIX_PATH` to `ON` unless the maintainer disable it.

And add an option `MORE_DEBUG_INFO` to write more debug info to the logs.

### CODE REVIEW

**Comment by JackBoosY** [comment]
I think we also need to handle `PKG_CONFIG_USE_CMAKE_PREFIX_PATH` in our vcpkg toolchain to fix the pkgconfig usage issue. cc @dg0yt

File: ports/vcpkg-cmake/vcpkg_cmake_configure.cmake [hunk 1]
```diff
@@ -193,6 +193,14 @@ function(vcpkg_cmake_configure)
     if(DEFINED VCPKG_CMAKE_CONFIGURE_OPTIONS_DEBUG)
         vcpkg_list(APPEND arg_OPTIONS_DEBUG ${VCPKG_CMAKE_CONFIGURE_OPTIONS_DEBUG})
     endif()
+    
+    if (NOT arg_DISABLE_DEFAULT_PKGCONFIG_PATH)
```
Comment (JackBoosY) [inline ports/vcpkg-cmake/vcpkg_cmake_configure.cmake:197]: Any better name?

Comment (dg0yt) [reply ports/vcpkg-cmake/vcpkg_cmake_configure.cmake:197]: "DEFAULT" is quite broad and might be misunderstood as the tool's/system's default.
I think it is enough to be able to overwrite it by passing `-DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=OFF` in OPTIONS.
I'm more concerned about whether it should be the default of the `vcpkg.cmake` toolchain file, to satisfy `find_dependency` requirements.

File: ports/vcpkg-cmake/vcpkg_cmake_configure.cmake [hunk 2]
```diff
@@ -193,6 +193,14 @@ function(vcpkg_cmake_configure)
...
+        vcpkg_list(APPEND arg_OPTIONS -DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=ON)
+    endif()
+    
+    if (NOT arg_MORE_DEBUG_INFO)
```
Comment (dg0yt) [inline ports/vcpkg-cmake/vcpkg_cmake_configure.cmake:201]: "DEBUG" can be misinterpreted as affecting the binaries to be built.
`MORE_DEBUG_INFO` will probably never be accepted for a port in the main repository.
`--trace-expand` can be passed in `OPTIONS`.
From that POV, I think it shouldn't be added unless it is meant to become a general option across build systems.

**Comment by Neumann-A** [comment]
Just patch in `cmake_minimum_required()` instead of adding options to `vcpkg-cmake` which are **rarely** needed and can be passed via `OPTIONS`?

**Comment by JackBoosY** [comment]
Wow our artifacts are too huge:
image

**Comment by JackBoosY** [comment]
My bad. It should be  `if (arg_MORE_DEBUG_INFO)`.

File: ports/vcpkg-cmake/vcpkg_cmake_configure.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/vcpkg-cmake/vcpkg_cmake_configure.cmake]: Given cmake's poor documentation, the `--trace` command may not be known to contributors.
This option is intended to provide better debugging option to contributors. I have changed its name and added a warning message.

**Comment by JackBoosY** [comment]
> Just patch in `cmake_minimum_required()` instead of adding options to `vcpkg-cmake` which are **rarely** needed and can be passed via `OPTIONS`?

This kind of issue is hard to find, which is why I have to force it: to avoid spending more time on this kind of issue.
Even if upstream cmake requires version lower than 3.1, the version of cmake used in vcpkg will still be much higher than 3.1, so I don't think there will be any problem with setting it to ON forever.

**Comment by dg0yt** [comment]
> Just patch in `cmake_minimum_required()` instead of adding options to `vcpkg-cmake` which are **rarely** needed and can be passed via `OPTIONS`?

I don't think this patching is acceptable.
It means to patch nearly every cmake port.
And it may break builds or usage because it changes active CMake policies to versions not tested upstream. This is against the purpose of CMake policies.

File: ports/vcpkg-cmake/vcpkg_cmake_configure.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/vcpkg-cmake/vcpkg_cmake_configure.cmake]: Well, you just explain why I would accept it as a general option (if there is something similar for meson etc.)
But adding another way to do this for CMake only just makes things more complex.

Comment (Neumann-A) [reply ports/vcpkg-cmake/vcpkg_cmake_configure.cmake]: > Given cmake's poor documentation, the --trace command may not be known to contributors.

RTM? It also shows up with `cmake --help` .....   So if you really want to help contributors here write a section on how to debug common port build failures instead ;)

**Comment by Neumann-A** [comment]
> It means to patch nearly every cmake port.

Well no. How many ports are there without `cmake_minimum_required()`? 

> And it may break builds or usage because it changes active CMake policies to versions not tested upstream. This is against the purpose of CMake policies.

welp. Newer CMake already breaks upstream testing so that is not really an argument. CMake is careful to not break legacy but behavior certainly already changed to when it was tested with an old cmake. Especially with stuff where the OLD behavior is deprecated and there is a warning that it will get removed.

**Comment by dg0yt** [comment]
> > It means to patch nearly every cmake port.
> 
> Well no. How many ports are there without `cmake_minimum_required()`?

Okay, the issue is mitigated by being enabled from CMake 3.1. This is good enough for the per-port approach. Although there is no need to patch if it can be solved with an OPTION.

But it is not enough to have `cmake_minimum_required`. It must be at least `VERSION 3.1`. I still stumble over ports which use 2.8 or less. Unmaintained upstreams don't go away in vcpkg.

**Comment by Neumann-A** [comment]
> But it is not enough to have cmake_minimum_required. It must be at least VERSION 3.1. I still stumble over ports which use 2.8 or less. Unmaintained upstreams don't go away in vcpkg.

be aware that some upstream add `cmake_minimum_required` later without knowing that it **needs to be the first line** in the `CMakeLists.txt`. If I have a problem with a port without `cmake_minimum_required` I just add it with 3.1 or higher to see if it fixes whatever error I observed ;) 

> Unmaintained upstreams don't go away in vcpkg.

Yeah but we can at least fix/patch them to work (somehow).

**Comment by dg0yt** [comment]
> be aware that some upstream add `cmake_minimum_required` later without knowing that it **needs to be the first line** in the `CMakeLists.txt`

Not the first line but before `project`. That is, for policies. I'm not sure if this is also required for var `CMAKE_MINIMUM_REQUIRED_VERSION`.

**Comment by JackBoosY** [comment]
libproxy:
```
-- Checking for modules 'gio-2.0>=2.26;gobject-2.0'
--   Found gio-2.0, version 2.70.5
--   Found gobject-2.0, version 2.70.5
```
```
[14/21] : && /usr/bin/c++ -fvisibility=hidden -fPIC -g -rdynamic libproxy/CMakeFiles/pxgsettings.dir/modules/pxgsettings.cpp.o -o libproxy/pxgsettings -L/mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib -Wl,-rpath,/mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib:/mnt/vcpkg-ci/installed/x64-linux/debug/lib:  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodman.so  -lgio-2.0  -lresolv  -lgmodule-2.0  -ldl  -lz  -lgobject-2.0  -lglib-2.0  -lm  -lpcre  -lffi && :
FAILED: libproxy/pxgsettings 
: && /usr/bin/c++ -fvisibility=hidden -fPIC -g -rdynamic libproxy/CMakeFiles/pxgsettings.dir/modules/pxgsettings.cpp.o -o libproxy/pxgsettings -L/mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib -Wl,-rpath,/mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib:/mnt/vcpkg-ci/installed/x64-linux/debug/lib:  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodman.so  -lgio-2.0  -lresolv  -lgmodule-2.0  -ldl  -lz  -lgobject-2.0  -lglib-2.0  -lm  -lpcre  -lffi && :
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gmain.c.o): in function `g_get_worker_context':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gmain.c:6164: undefined reference to `pthread_sigmask'
/usr/bin/ld: /mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gmain.c:6169: undefined reference to `pthread_sigmask'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rec_mutex_impl_new':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:288: undefined reference to `pthread_mutexattr_init'
/usr/bin/ld: /mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:289: undefined reference to `pthread_mutexattr_settype'
/usr/bin/ld: /mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:291: undefined reference to `pthread_mutexattr_destroy'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rec_mutex_trylock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:432: undefined reference to `pthread_mutex_trylock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_impl_new':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:450: undefined reference to `pthread_rwlock_init'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_impl_free':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:459: undefined reference to `pthread_rwlock_destroy'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_writer_lock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:552: undefined reference to `pthread_rwlock_wrlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_writer_trylock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:574: undefined reference to `pthread_rwlock_trywrlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_writer_unlock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:594: undefined reference to `pthread_rwlock_unlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_reader_lock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:622: undefined reference to `pthread_rwlock_rdlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_reader_trylock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:643: undefined reference to `pthread_rwlock_tryrdlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_rw_lock_reader_unlock':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:663: undefined reference to `pthread_rwlock_unlock'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_private_impl_new':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1030: undefined reference to `pthread_key_create'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_private_impl_free':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1042: undefined reference to `pthread_key_delete'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_private_get':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1082: undefined reference to `pthread_getspecific'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_private_set':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1102: undefined reference to `pthread_setspecific'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_private_replace':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1128: undefined reference to `pthread_getspecific'
/usr/bin/ld: /mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1130: undefined reference to `pthread_setspecific'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_system_thread_free':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1169: undefined reference to `pthread_detach'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_system_thread_new':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1310: undefined reference to `pthread_attr_setstacksize'
/usr/bin/ld: /mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1323: undefined reference to `pthread_create'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_system_thread_wait':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1369: undefined reference to `pthread_join'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a(gthread-posix.c.o): in function `g_system_thread_set_name':
/mnt/vcpkg-ci/buildtrees/glib/x64-linux-dbg/../src/2.70.5-fa6b1e9a0f.clean/glib/gthread-posix.c:1388: undefined reference to `pthread_setname_np'
collect2: error: ld returned 1 exit status
```

hyperscan:
```
-- Checking for module 'libpcre>=8.41'
--   Found libpcre, version 8.45
```
```
[333/338] cmd.exe /C "cd . && D:\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=tools\hscollider\CMakeFiles\hscollider.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1432~1.313\bin\Hostx64\x86\link.exe  tools\hscollider\CMakeFiles\hscollider.dir\Corpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\FileCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\ColliderCorporaParser.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\NfaGeneratedCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GraphTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GroundTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\UltimateTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\args.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\limit.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\pcre_util.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\sig.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\Thread.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\main.cpp.obj  /out:bin\hscollider.exe /implib:lib\hscollider.lib /pdb:bin\hscollider.pdb /version:0.0 /machine:X86 /nologo    /debug /INCREMENTAL /subsystem:console /STACK:8388608,8388608  lib\hs.lib  pcre.lib  lib\databaseutil.lib  lib\expressionutil.lib  lib\corpusomatic.lib  lib\crosscompileutil.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
FAILED: bin/hscollider.exe 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=tools\hscollider\CMakeFiles\hscollider.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1432~1.313\bin\Hostx64\x86\link.exe  tools\hscollider\CMakeFiles\hscollider.dir\Corpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\FileCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\ColliderCorporaParser.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\NfaGeneratedCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GraphTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GroundTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\UltimateTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\args.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\limit.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\pcre_util.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\sig.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\Thread.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\main.cpp.obj  /out:bin\hscollider.exe /implib:lib\hscollider.lib /pdb:bin\hscollider.pdb /version:0.0 /machine:X86 /nologo    /debug /INCREMENTAL /subsystem:console /STACK:8388608,8388608  lib\hs.lib  pcre.lib  lib\databaseutil.lib  lib\expressionutil.lib  lib\corpusomatic.lib  lib\crosscompileutil.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1432~1.313\bin\Hostx64\x86\link.exe tools\hscollider\CMakeFiles\hscollider.dir\Corpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\FileCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\ColliderCorporaParser.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\NfaGeneratedCorpora.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GraphTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\GroundTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\UltimateTruth.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\args.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\limit.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\pcre_util.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\sig.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\Thread.cpp.obj tools\hscollider\CMakeFiles\hscollider.dir\main.cpp.obj /out:bin\hscollider.exe /implib:lib\hscollider.lib /pdb:bin\hscollider.pdb /version:0.0 /machine:X86 /nologo /debug /INCREMENTAL /subsystem:console /STACK:8388608,8388608 lib\hs.lib pcre.lib lib\databaseutil.lib lib\expressionutil.lib lib\corpusomatic.lib lib\crosscompileutil.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:tools\hscollider\CMakeFiles\hscollider.dir/intermediate.manifest tools\hscollider\CMakeFiles\hscollider.dir/manifest.res" failed (exit code 1104) with the following output:
LINK : fatal error LNK1104: cannot open file 'pcre.lib'
```

libfreenect2:
```
-- Checking for module 'glfw3'
--   Found glfw3, version 3.3.7
-- Found OpenGL: opengl32   
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Failed
-- Performing Test COMPILER_HAS_DEPRECATED
-- Performing Test COMPILER_HAS_DEPRECATED - Success
-- Linking with these libraries: 
 optimized
 D:/installed/x86-windows/lib/libusb-1.0.lib
 debug
 D:/installed/x86-windows/debug/lib/libusb-1.0.lib
 D:/installed/x86-windows/lib/turbojpeg.lib
 GLFW3_LIBRARY-NOTFOUND
 opengl32
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Feature list:
--   CUDA    disabled
--   CXX11    disabled
--   Examples    disabled
--   OpenCL    disabled
--   OpenGL    yes
--   OpenNI2    disabled
--   TegraJPEG    no
--   Threading    tinythread
--   TurboJPEG    yes
--   VAAPI    no
--   VideoToolbox    no (Apple only)
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
GLFW3_LIBRARY
    linked by target "freenect2" in directory D:/buildtrees/libfreenect2/src/v0.2.0-bc6450abec.clean
```

**Comment by ras0219-msft** [comment]
I can think of no reason that a port shouldn't consider `$installed/lib/pkgconfig`/`$installed/share/pkgconfig`, so I don't see a need to turn this off. Additionally, as @dg0yt points out, we need end users to also look in these paths regardless of their (mis)behavior.

I see two reasonable ways forward that accomplish the goals:
1. Set `PKG_CONFIG_USE_CMAKE_PREFIX_PATH` to `ON` always for everyone (in vcpkg.cmake)
2. Append vcpkg's paths to `$PKG_CONFIG_PATH` within the CMake build if they aren't present (also in vcpkg.cmake)

The second option avoids undesired additional behaviors -- we don't want to inject non-vcpkg search paths that the user doesn't want.

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please ping me to reopen if work is still being done.

**Pull request closed without merging**